projects/plugins/jetpack/modules/masterbar/admin-menu/admin-menu.js

```diff
@@ -58,16 +58,42 @@
 					}, 50 );
 				} );
 			}
+
+			const jitmDismissButton = adminMenu.querySelector( '.banner__dismiss' );
+			if ( jitmDismissButton ) {
+				jitmDismissButton.addEventListener( 'click', function ( event ) {
+					event.preventDefault();
+
+					const siteNotice = document.getElementById( 'toplevel_page_site-notices' );
+					if ( siteNotice ) {
+						siteNotice.style.display = 'none';
+					}
+
+					makeAjaxRequest(
+						'POST',
+						window.jitm_config.api_root + 'jetpack/v4/jitm',
+						'application/json',
+						JSON.stringify( {
+							id: event.target.parentNode.dataset.feature_id,
+							feature_class: event.target.parentNode.dataset.feature_class,
+						} ),
+						window.jitm_config.nonce
+					);
```

Comment:
This seems to trigger a CORS error for me (tested on a Simple site), and the dismissal status doesn't persist.

Reply:
@BogdanUngureanu has messaged me about an issue with `window.jitm_config`  not existing on simple sites. I expect the same has happened here and the CORS error is a failed attempt at recording the JS error.

Reply:
On atomic the jetpack JITM class is used on the page request, which means `window.jitm_config` is created and so when `class-wpcom-admin-menu.php` does JITM directly with jitm-engine, the dismissal works because `window.jitm_config` has already been set-up 

Simple sites don't use the jetpack JITM class, so `window.jitm_config` isn't setup.

The only parts of `window.jitm_config` that matters here is:
 * api_root - which is just `rest_url`
 * nonce - which is just the nonce for wp_rest

There's probably a better way to handle this - either by making the request via the wp js library, which presumably already handles this somehow already, or just by setting data attribs on the notice HTML and having the JS use that.

Reply:
What if for simple sites we make a request server-side to the same endpoint called by Calypso? It'd be something similar to what we do to store the collapse status of the sidebar on Atomic sites:
- Click on the dismiss button triggers an AJAX request to `${ ajaxurl }?action=dismiss_upsell_nudge&feature_class=${ feature_class }&id=${ id }` ([see collapse status counterpart)](https://github.com/Automattic/jetpack/blob/3d30d7a43d9bd5f6f80dddb64d45b505e221867e/projects/plugins/jetpack/modules/masterbar/admin-menu/admin-menu.js#L75-L82).
- We register an AJAX action that handles these requests ([see collapse status counterpart](https://github.com/Automattic/jetpack/blob/3d30d7a43d9bd5f6f80dddb64d45b505e221867e/projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php#L28)).
- The action makes a remote request to the `/jetpack-blogs/${ siteID }/rest-api` endpoint using [the same parameters set by Calypso](https://github.com/Automattic/wp-calypso/blob/1983cfdfc17294329a3f0663f6acd6da1994cf71/client/state/data-layer/wpcom/sites/jitm/index.js#L84-L94) ([see collapse status counterpart](https://github.com/Automattic/jetpack/blob/3d30d7a43d9bd5f6f80dddb64d45b505e221867e/projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php#L388-L401)).

That should work for Atomic sites as well, but we might want to optimize it by making the request directly to the `/jetpack/v4/jitm` endpoint.

Reply:
@mmtr thank you! I had implemented something using the client-side proxy, but I ran into some issues and the solution you've outlined above seems more robust, so I'm switching to doing that.

One question though - I think we should either always use the `/jetpack-blogs/${ siteID }/rest-api` endpoint if the abstraction is worthwhile or never use it if it's preferable to optimize it away.

As the ajax action will be used only on wpcom simple and atomic sites (**not jetpack sites** - they don't use nav-unification and don't implement JITMs in their _Admin_Menu class) then the call to `/jetpack-blogs/${ siteID }/rest-api` will always end up going via `Jetpack_Direct_Call::dismiss_jitm`, so the overhead is minimal

My brain is a bit fuzzy today, does that make sense or am I missing something?

Reply:
> if the abstraction is worthwhile 

I think it is! Otherwise Calypso would use the `/jetpack/v4/jitm` endpoint directly. Let's try a unified approach then, a use always the `/jetpack-blogs/${ siteID }/rest-api` endpoint for both simple and atomic.

Reply:
In the end I just went with the JITM classes, which for atomic will use any necessary REST APIs and for simple avoids the indirection - which when I quickly tested didn't work and was hard to debug. This mirrors the approach that retrieving the JITM content in this class takes.